rules/ast/functions.go

```diff
@@ -498,6 +499,43 @@ func derivImpl(timestamp clientmodel.Timestamp, args []Node) interface{} {
 	return resultVector
 }
 
+// === histogram_quantile(k ScalarNode, vector VectorNode) Vector ===
```

Comment:
> why would 1. error out instead of just creating a one-element vector with the overall quantile? 

So that it has consistent behaviour with by clauses in other cases. Your proposal is doing magic behind the scenes that'll cause confusion when users try to generalise to the behaviours of similar functions.

For example `topk` has the same look (takes in an instant vector, returns an instant vector) - and it operates over all of the input timeseries as one. If you wanted it to work over a subset you'd have to add a by clause. The aggregation functions also work this way. I propose that histogram_quantile requires the same, if you want to act on something other than everything you need to provide a by clause.